### PR TITLE
chore: add multi-locale date picker to components

### PR DESCRIPTION
**Pull request by ghost**
1. Set baseUrl of mui package tsconfig file to "." in order to use absolute path for imports
2. Apply some change to default-mui-picker-localization component
3. Add OmitAndReplace type mapping in utils package

### CODE REVIEW

File: packages/mui/src/date-picker/px-multi-locale-date-picker.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+export function MultiLocaleDatePicker<In, Out = In>(props: Props<In, Out>) {
+  const [locale, setLocale] = useState<Locale>(Locale.defaultLocale);
+  const handleLocaleChange = () => {
+    setLocale((prevLocale) =>
+      prevLocale === Locale.en ? Locale.fa : Locale.en,
+    );
+  };
+
+  return (
+    <DefaultMuiPickerLocalization locale={locale}>
```
Comment (mh-alahdadian) [inline packages/mui/src/date-picker/px-multi-locale-date-picker.tsx]: use LocalizationProvider directly

File: packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  const localeText = useLocaleText();
+
+  return (
+    <DialogActionBar>
+      <PickersActionBar {...pickersActionBarProps} />
+      <DialogActions>
+        <Button onClick={onLocaleChange}>
+          {localeText.changeLocaleButtonLabel &&
+            localeText.changeLocaleButtonLabel[locale]}
```
Comment (mh-alahdadian) [inline packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx]: localeText.changeLocaleButtonLabel?.[locale]

File: packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  const { locale, onLocaleChange, ...pickersActionBarProps } = props;
+
+  const localeText = useLocaleText();
+
+  return (
+    <DialogActionBar>
+      <PickersActionBar {...pickersActionBarProps} />
+      <DialogActions>
+        <Button onClick={onLocaleChange}>
+          {localeText.changeLocaleButtonLabel &&
```
Comment (mh-alahdadian) [inline packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx]: remove

File: packages/mui/src/pickers-common/default-mui-picker-localization.tsx [hunk 4]
```diff
@@ -7,6 +7,8 @@ import { Locale } from '../constant-types';
 
 interface Props {
   locale: Locale;
+  lang?: Locale;
+  localeTexts?: Partial<PickersLocaleText>;
```
Comment (mh-alahdadian) [inline packages/mui/src/pickers-common/default-mui-picker-localization.tsx]: remove

File: packages/mui/src/stories/decorators/calendarDecorator.tsx [hunk 5]
```diff
@@ -1,14 +1,16 @@
...
 export const calendarDecorator =
-  (lang?: 'en' | 'fa'): DecoratorFunction<StoryFnReactReturnType> =>
+  (locale?: Locale): DecoratorFunction<StoryFnReactReturnType> =>
   (Story, context) => {
-    const locale = lang ?? context?.globals.locale ?? 'fa';
-
     return (
-      <DefaultMuiPickerLocalization locale={locale}>
+      <DefaultMuiPickerLocalization
+        locale={locale || Locale.defaultLocale}
```
Comment (mh-alahdadian) [inline packages/mui/src/stories/decorators/calendarDecorator.tsx:11]: undo

**Review by mh-alahdadian — CHANGES_REQUESTED** [review]

File: packages/mui/src/date-picker/px-multi-locale-date-picker.tsx (hunk 1 above)
Comment (ghost) [reply packages/mui/src/date-picker/px-multi-locale-date-picker.tsx]: Done!

File: packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx (hunk 2 above)
Comment (ghost) [reply packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx]: Done!

File: packages/mui/src/pickers-common/default-mui-picker-localization.tsx (hunk 4 above)
Comment (ghost) [reply packages/mui/src/pickers-common/default-mui-picker-localization.tsx]: I differentiated between lang and locale.
locale determines calendar system and lang determines the language used for locale text and other labels!

File: packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx (hunk 3 above)
Comment (ghost) [reply packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx]: remove what?

Comment (AmirABody) [reply packages/mui/src/pickers-common/MultiLocalePickersActionBar.tsx]: remove what?

File: packages/mui/src/pickers-common/default-mui-picker-localization.tsx [hunk 6]
```diff
@@ -1,63 +1,36 @@
...
     <LocalizationProvider
-      dateAdapter={locale === 'en' ? DateFnsAdapter : DateFnsJalaliAdapter}
+      dateAdapter={getLocalizedDateFnsAdapter(locale)}
       localeText={
-        localeTexts || lang === 'en'
-          ? defaultEnglishLocaleTexts
-          : defaultPersianLocaleTexts
+        lang === 'en' ? defaultEnglishLocaleTexts : defaultPersianLocaleTexts
+      }
+      adapterLocale={
```
Comment (mh-alahdadian) [inline packages/mui/src/pickers-common/default-mui-picker-localization.tsx:25]: what is this for ?

File: packages/mui/src/stories/decorators/calendarDecorator.tsx (hunk 5 above)
Comment (ghost) [reply packages/mui/src/stories/decorators/calendarDecorator.tsx:11]: That's better I think!

File: packages/mui/src/pickers-common/default-mui-picker-localization.tsx [hunk 7]
```diff
@@ -1,63 +1,36 @@
...
+        locale === Locale.en
+          ? require('date-fns/locale')[lang === Locale.en ? 'enUS' : 'faIR']
```
Comment (mh-alahdadian) [inline packages/mui/src/pickers-common/default-mui-picker-localization.tsx:27]: redundant condition

File: packages/mui/src/stories/stories/date-picker.stories.tsx [hunk 8]
```diff
@@ -49,6 +49,7 @@ export const JalaliCalender: Story = Template.bind({});
 JalaliCalender.decorators = [calendarDecorator(Locale.fa)];
 
 export const MultiLocaleCalendar: Story = MultiLocaleTemplate.bind({});
+MultiLocaleCalendar.decorators = [calendarDecorator()];
```
Comment (mh-alahdadian) [inline packages/mui/src/stories/stories/date-picker.stories.tsx:52]: is this necessary ? I think it can be inherited

**Review by mh-alahdadian — CHANGES_REQUESTED** [review]

**Pull request closed without merging**